From e65598009c2ade68e60559f81f63a7a894cdadef Mon Sep 17 00:00:00 2001
From: Benjamin Reis <benjamin.reis@vates.tech>
Date: Mon, 30 Sep 2024 13:42:21 +0200
Subject: [PATCH] Bring up IPv6 only virtual PIF as well on startup

Like for IPv4 bring up IPv6 configured virtual PIF when
starting up an host.

Signed-off-by: Benjamin Reis <benjamin.reis@vates.tech>
---
 ocaml/xapi/xapi_pif.ml | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/ocaml/xapi/xapi_pif.ml b/ocaml/xapi/xapi_pif.ml
index 3df1d692b39..a2383ed9d9b 100644
--- a/ocaml/xapi/xapi_pif.ml
+++ b/ocaml/xapi/xapi_pif.ml
@@ -1109,7 +1109,12 @@ let calculate_pifs_required_at_start_of_day ~__context =
                        ( Not (Eq (Field "bond_master_of", Literal "()"))
                        , Eq (Field "physical", Literal "true")
                        )
-                   , Not (Eq (Field "ip_configuration_mode", Literal "None"))
+                   , Not
+                       (And
+                          ( Eq (Field "ip_configuration_mode", Literal "None")
+                          , Eq (Field "ipv6_configuration_mode", Literal "None")
+                          )
+                       )
                    )
                )
            )
